resource-managers/kubernetes/core/pom.xml

```diff
@@ -29,8 +29,25 @@
   <name>Spark Project Kubernetes</name>
   <properties>
     <sbt.project.name>kubernetes</sbt.project.name>
+    <volcano.exclude>**/*Volcano*.scala</volcano.exclude>
   </properties>
 
+  <profiles>
+    <profile>
+      <id>volcano</id>
```

Comment:
Hm, sorry if I missed but why should we create a profile?

Reply:
Yep, we had some discussion on: https://github.com/apache/spark/pull/35422#discussion_r800986656, as dongjoon suggestion:

```
Technically, Apache Spark uses three options when it embraces new features. I'd prefer (2). Shall we put this under -Pvolcano profile please?

1. External module like Kafka Streaming
2. Profile like HiveThriftServer or SparkR
3. Built-in module.
```

Reply:
Hm. I wonder if it's better to have it as built-in because the deps and codes it needs are pretty minor.

Reply:
Let me enumerate the difference between built in module and profile:
1. Built-in:
- `volcano-model-v1beta1` deps for spark-kubernetes module
- `volcano-client` deps for spark-integrations-test module

2. Profile:
- `volcano-model-v1beta1` deps for spark-kubernetes module with volcano profile
- `volcano-client` deps for spark-integrations-test module with volcano profile
- Add exclude info in `sbt` AND `maven`.
- Add `-Pvolcano` in everywhere.
- Required seprate file for volcano (including test/code) to help add profile filter in mvn/sbt, we couldn't put volcano test mix in existing test unless `extends`.

I have to admit profile module is really a little bit complex for me even I have made efforts to complete the separating work.

So if it possible, I personally suggest that have it as bulit-in module and still required to use the separate class / file for Volcano to reduce the impact. Of course, I'd like to hear some input from @dongjoon-hyun at this time. WDYT?

Reply:
I added my comments here.
- https://github.com/apache/spark/pull/35495#pullrequestreview-880891074
- https://github.com/apache/spark/pull/35495#issuecomment-1037480828
- https://github.com/apache/spark/pull/35495#issuecomment-1037484853